pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -53,6 +54,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.CacheResults, "query-frontend.cache-results", false, "Cache query results.")
 	f.BoolVar(&cfg.ShardedQueries, "query-frontend.parallelize-shardable-queries", false, "True to enable query sharding.")
 	f.BoolVar(&cfg.CacheUnalignedRequests, "query-frontend.cache-unaligned-requests", false, "Cache requests that are not step-aligned.")
+	f.Uint64Var(&cfg.MaxSeriesPerShard, "query-frontend.max-series-per-shard", 0, "How many series a single sharded sub-request should load at most. 0 to disable cardinality-based sharding.")
```

Comment:
Should this be `query-frontend.query-sharding-max-series-per-shard` so it aligns with `query-sharding-max-sharded-queries` and `query-sharding-total-shards`?

Reply:
Yes, I think it's a good idea.